tests/testAutopas/tests/selectors/ContainerSelectorTest.cpp

```diff
@@ -27,3 +31,104 @@ TEST_F(ContainerSelectorTest, testSelectAndGetCurrentContainer) {
+        for (auto z : getPossible1D(bBoxMin[2], bBoxMax[2])) {
+          std::array<double, 3> pos{x, y, z};
+          Particle p(pos, {0., 0., 0.}, id);
+          if (autopas::utils::inBox(pos, bBoxMin, bBoxMax)) {
+            container->addParticle(p);
+          } else {
+            if (autopas::utils::inBox(pos, autopas::ArrayMath::sub(bBoxMin, std::array<double, 3>{cutoff}),
+                                      autopas::ArrayMath::add(bBoxMin, std::array<double, 3>{cutoff})) or
+                autopas::utils::StringUtils::to_string(container->getContainerType()).find("Verlet") !=
+                    std::string::npos) {
+              container->addHaloParticle(p);
+            }
+          }
+          ++id;
+        }
+      }
+    }
+  }
+  std::vector<Particle> beforeListInner, beforeListHalo,
+      beforeListHaloVerletOnly /*for particles only in verlet containers*/;
+  for (auto iter = containerSelector.getCurrentContainer()->begin(autopas::IteratorBehavior::ownedOnly); iter.isValid();
+       ++iter) {
+    beforeListInner.push_back(*iter);
+  }
+  for (auto iter = containerSelector.getCurrentContainer()->begin(autopas::IteratorBehavior::haloOnly); iter.isValid();
+       ++iter) {
+    if (autopas::utils::inBox(iter->getR(), autopas::ArrayMath::sub(bBoxMin, std::array<double, 3>{cutoff}),
+                              autopas::ArrayMath::add(bBoxMin, std::array<double, 3>{cutoff}))) {
+      beforeListHalo.push_back(*iter);
+    } else {
+      beforeListHaloVerletOnly.push_back(*iter);
+    }
+  }
+
+  // select container to which we want to convert to
+  containerSelector.selectContainer(to);
+
+  std::vector<Particle> afterListInner, afterListHalo, afterListHaloVerletOnly;
+  for (auto iter = containerSelector.getCurrentContainer()->begin(autopas::IteratorBehavior::ownedOnly); iter.isValid();
+       ++iter) {
+    afterListInner.push_back(*iter);
+  }
+  for (auto iter = containerSelector.getCurrentContainer()->begin(autopas::IteratorBehavior::haloOnly); iter.isValid();
+       ++iter) {
+    if (autopas::utils::inBox(iter->getR(), autopas::ArrayMath::sub(bBoxMin, std::array<double, 3>{cutoff}),
+                              autopas::ArrayMath::add(bBoxMin, std::array<double, 3>{cutoff}))) {
+      afterListHalo.push_back(*iter);
+    } else {
+      afterListHaloVerletOnly.push_back(*iter);
+    }
+  }
+
+  EXPECT_THAT(afterListInner, UnorderedElementsAreArray(beforeListInner));
+  EXPECT_THAT(afterListHalo, UnorderedElementsAreArray(beforeListHalo));
+  if (autopas::utils::StringUtils::to_string(to).find("Verlet") != std::string::npos and
+      autopas::utils::StringUtils::to_string(from).find("Verlet") != std::string::npos) {
+    EXPECT_THAT(afterListHaloVerletOnly, UnorderedElementsAreArray(beforeListHaloVerletOnly));
+  }
+}
+
+INSTANTIATE_TEST_SUITE_P(
+    Generated, ContainerSelectorTest,
+    Combine(ValuesIn([]() -> std::vector<autopas::ContainerOption> {
```

Comment:
add a TODO that VerletClusterLists should be covered by this test if <insertListOfProblemsHere> is completed